File: src/decoders/account-kind.ts
import { AccountKind, UnreachableError } from "~/models";

/**
 * @param path mobile.eleve.html or eleve.html, both works.
 */
export const decodeAccountKindFromPath = (path: string): AccountKind => {
  const segments = path.split(".");
  segments.pop(); // remove .html

  switch (segments.pop()) {
    case "eleve": return AccountKind.STUDENT;
    case "parent": return AccountKind.PARENT;
    case "professeur": return AccountKind.TEACHER;
    default: throw new UnreachableError("decodeAccountKindFromPath");
  }
};

File: library/src/commonMain/kotlin/decoders/account-kind.kt
package decoders

import models.AccountKind
import models.errors.UnreachableError

/**
 * @param path mobile.eleve.html or eleve.html, both works.
 */
fun decodeAccountKindFromPath (path: String): AccountKind {
    val segments = path.split('.').toMutableList()
    segments.removeLast()

    when (segments.removeLast()) {
        "eleve" -> return AccountKind.STUDENT
        "parent" -> return AccountKind.PARENT
        "professeur" -> return AccountKind.TEACHER
    }

    throw UnreachableError("decodeAccountKindFromPath")
}